### ISSUE TITLE
Attach comments to enum values

### ISSUE THREAD

**Issue by marcjulian** [prisma/prisma#3807]
## Problem

```prisma
/// user role
enum Role {
  ADMIN /// allowed to do everything
  USER
}
```

Comment `allowed to do everything` is not attached to the ADMIN value in the DMMF generator.

<details>
<summary>Generator DMMF output</summary>

```prisma
// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

datasource db {
  provider = "postgres"
  url      = env("POSTGRESQL_URL")
}

generator client {
  provider = "prisma-client-js"
}

generator dbml {
  provider = "node ./dist/generator.js"
}

model User {
  id        Int      @id @default(autoincrement())
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
  email     String   @unique
  name      String?
  posts     Post[]
  profile   Profile?
  /// user role
  role      Role     @default(USER)
}

/// User profile
model Profile {
  id     Int     @default(autoincrement()) @id
  bio    String?
  user   User    @relation(fields: [userId], references: [id])
  userId Int     @unique
}

model Post {
  id         Int        @id @default(autoincrement())
  title      String     @default("")
  content    String?
  published  Boolean    @default(false)
  author     User?      @relation(fields: [authorId], references: [id])
  authorId   Int?
  categories Category[]
}

model Category {
  id    Int    @id @default(autoincrement())
  name  String
  posts Post[]
}

/// user role
enum Role {
  ADMIN /// allowed to do everything
  USER
}
```

```json
{
  "enums": [
    {
      "name": "Role",
      "values": [
        { "name": "ADMIN", "dbName": null },
        { "name": "USER", "dbName": null }
      ],
      "dbName": null,
      "documentation": "user role"
    }
  ],
  "models": [
    {
      "name": "User",
      "isEmbedded": false,
      "dbName": null,
      "fields": [
        {
          "name": "id",
          "kind": "scalar",
          "isList": false,
          "isRequired": true,
          "isUnique": false,
          "isId": true,
          "isReadOnly": false,
          "type": "Int",
          "hasDefaultValue": true,
          "default": { "name": "autoincrement", "args": [] },
          "isGenerated": false,
          "isUpdatedAt": false
        },
        {
          "name": "createdAt",
          "kind": "scalar",
          "isList": false,
          "isRequired": true,
          "isUnique": false,
          "isId": false,
          "isReadOnly": false,
          "type": "DateTime",
          "hasDefaultValue": true,
          "default": { "name": "now", "args": [] },
          "isGenerated": false,
          "isUpdatedAt": false
        },
        {
          "name": "updatedAt",
          "kind": "scalar",
          "isList": false,
          "isRequired": true,
          "isUnique": false,
          "isId": false,
          "isReadOnly": false,
          "type": "DateTime",
          "hasDefaultValue": false,
          "isGenerated": false,
          "isUpdatedAt": true
        },
        {
          "name": "email",
          "kind": "scalar",
          "isList": false,
          "isRequired": true,
          "isUnique": true,
          "isId": false,
          "isReadOnly": false,
          "type": "String",
          "hasDefaultValue": false,
          "isGenerated": false,
          "isUpdatedAt": false
        },
        {
          "name": "name",
          "kind": "scalar",
          "isList": false,
          "isRequired": false,
          "isUnique": false,
          "isId": false,
          "isReadOnly": false,
          "type": "String",
          "hasDefaultValue": false,
          "isGenerated": false,
          "isUpdatedAt": false
        },
        {
          "name": "posts",
          "kind": "object",
          "isList": true,
          "isRequired": false,
          "isUnique": false,
          "isId": false,
          "isReadOnly": false,
          "type": "Post",
          "hasDefaultValue": false,
          "relationName": "PostToUser",
          "relationFromFields": [],
          "relationToFields": [],
          "relationOnDelete": "NONE",
          "isGenerated": false,
          "isUpdatedAt": false
        },
        {
          "name": "profile",
          "kind": "object",
          "isList": false,
          "isRequired": false,
          "isUnique": false,
          "isId": false,
          "isReadOnly": false,
          "type": "Profile",
          "hasDefaultValue": false,
          "relationName": "ProfileToUser",
          "relationFromFields": [],
          "relationToFields": [],
          "relationOnDelete": "NONE",
          "isGenerated": false,
          "isUpdatedAt": false
        },
        {
          "name": "role",
          "kind": "enum",
          "isList": false,
          "isRequired": true,
          "isUnique": false,
          "isId": false,
          "isReadOnly": false,
          "type": "Role",
          "hasDefaultValue": true,
          "default": "USER",
          "isGenerated": false,
          "isUpdatedAt": false,
          "documentation": "user role"
        }
      ],
      "isGenerated": false,
      "idFields": [],
      "uniqueFields": [],
      "uniqueIndexes": []
    },
   ...
  ]
}
```
</details>

## Suggested solution

Attach comments to enum values

```diff
interface EnumValue {
   name: string;
   dbName: string;
+  documentation?: string;
}
interface DatamodelEnum {
  name: string;
  values: EnumValue[];
  dbName?: string | null;
  documentation?: string;
}
```

**Comment by SynergyEvolved** [comment]
Hi There,

I am using prisma-nestjs-graphql generator and this is stopping the comments from the schema showing up in the generated registerEnumType.

https://github.com/unlight/prisma-nestjs-graphql/issues/43#issuecomment-904200469

**Comment by dieeisenefaust** [comment]
In a broader sense, I would actually also ask that the triple-slash comments ("///") also be attached for composite types. 

Currently comments only appear to attach to actual model types.

Seems related to me, but please let me know if you would rather I open a separate ticket.

**Comment by janpio** [comment]
Separate issue please @dieeisenefaust - that is just something we overlooked.

**Comment by dieeisenefaust** [comment]
Issue #13726 has been opened for composite type comments.

**Comment by tomhoule** [comment]
Since https://github.com/prisma/prisma-engines/pull/3034, we test that we do parse these comments and attach them to the relevant enum value. I'll hand over to the client team and let them decide whether they want to add that documentation field to DMMF.

**Comment by michaelpoellath** [comment]
Fixed with https://github.com/prisma/prisma-engines/pull/4037

**Comment by janpio** [comment]
This should be available in the current `@dev` versions already, so `prisma@dev` and `@prisma/client@dev`. If you have time, please temporarily install these and double-check if the missing documentation is now available as expected. This will be included in `prisma@4.16.0` next Tuesday as well 🚀
